src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -400,6 +401,9 @@ protected Query doToQuery(QueryShardContext context) {
         // We need transform distance radius to right type of engine required radius.
         Float radius = null;
         if (this.distance != null) {
+            if (this.distance < 0 && !SpaceType.INNER_PRODUCT.equals(spaceType)) {
+                throw new IllegalArgumentException("[" + NAME + "] requires distance to be non-negative for space type: " + spaceType);
+            }
```

Comment:
add a TODO that this condition will be removed in next PR.

Reply:
What's the point to remove? I think here can also be used after supporting Lucene IP distance.

Reply:
I think I didn't read `!` in the condition.